controllers/integrationpipeline/integrationpipeline_controller.go

```diff
@@ -85,6 +87,13 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, err
 	}
 
+	if helpers.HasPipelineRunFinished(pipelineRun) && pipelineRun.GetDeletionTimestamp() != nil &&
```

Comment:
Why we don't handle the issue on the place where it occurs instead of adding triple condition into reconciler, that may not even catch all issues?

Reply:
I was thinking about adding the fix at the place where it occurs but it felt like a bandage for an injury we're not supposed to have in the first place. I wondered if we even need to process these kinds of PLRs caught by the above `if` condition and the answer was no. But I'm curious about what you think about this.

Reply:
sorry couldn't find time earlier,  but it looks much better